python_ta/debug/accumulation_table.py

```diff
@@ -46,9 +46,11 @@ def get_loop_node(frame: types.FrameType) -> Union[astroid.For, astroid.While]:
     lst_str_lines = func_string.splitlines()
     lst_from_with_stmt = lst_str_lines[with_stmt_index + 1 :]
     num_whitespace = num_whitespaces(lst_str_lines[with_stmt_index])
-    loop_lines = get_loop_lines(lst_from_with_stmt, num_whitespace)
+    with_lines = get_with_lines(lst_from_with_stmt, num_whitespace)
 
-    return astroid.parse(loop_lines).body[0]
+    for node in astroid.parse(with_lines).body:
```

Comment:
I realized that we can extend this even further by using the astroid `nodes_of_class` method. `node.nodes_of_class((astroid.For, astroid.While))` returns a _generator_ for the nodes of those types contained in `node`.

You can call `next` on a generator to obtain the first item.

This will allow `AccumulationTable` to detect loops that aren't top-level statements in the `with`, but nested (e.g. within an if statement).

```python
with AccumulationTable(...):
    if something:
        for x in numbers:
            ...
```

Please add some tests for this as well!